src/node.js

```diff
@@ -130,21 +130,14 @@
         // If -i or --interactive were passed, or stdin is a TTY.
         if (process._forceRepl || NativeModule.require('tty').isatty(0)) {
           // REPL
-          var opts = {
-            useGlobal: true,
-            ignoreUndefined: false
-          };
-          if (parseInt(process.env['NODE_NO_READLINE'], 10)) {
-            opts.terminal = false;
-          }
-          if (parseInt(process.env['NODE_DISABLE_COLORS'], 10)) {
-            opts.useColors = false;
-          }
-          var repl = Module.requireRepl().start(opts);
-          repl.on('exit', function() {
-            process.exit();
+          Module.requireRepl().createRepl(process.env, function(err, repl) {
+            if (err) {
+              throw err;
```

Comment:
I want to check on this behavior. If there's an error, it's most likely because the history file is corrupted or otherwise inaccessible. As written, this crashes on those errors. However, it's possible that we can retry with history disabled, and mention that there's a problem with the file (& what that problem is!)